app/middleware/cors.py

```diff
@@ -0,0 +1,44 @@
+import logging
+import re
+from urllib.parse import urlparse
+
+from django.conf import settings
+from django.utils.cache import add_never_cache_headers
+from django.utils.cache import patch_cache_control
+from django.utils.cache import patch_response_headers
+
+logger = logging.getLogger(__name__)
+
+STAC_BASE = settings.STAC_BASE
+STAC_BASE_V = settings.STAC_BASE_V
+
+
+class CORSHeadersMiddleware:
+    '''Middleware that adds appropriate CORS headers.
+
+    CORS has only effect on browser applications (e.g. STAC browser),
+    not on other systems. Therefore we only allow GET and HEAD requests
+    on all endpoints except /search.
```

Comment:
How about the get-token endpoint? Probably better to not allow POST from origin "*" though..

Reply:
We would only need to allow CORS for POST get-token if this was done/issued directly within a browser that has cors protection enabled. I assume this is not the case, so we leave it out.